services/payments/datastore.go

```diff
@@ -147,25 +218,25 @@ func (t *Transaction) SignTransaction(ctx context.Context, kmsClient wrappedKMSC
 }
 
 // MarshalJSON - custom marshaling of transaction type.
-func (t *Transaction) MarshalJSON() ([]byte, error) {
-	type Alias Transaction
+func (a *AuthenticatedPaymentState) MarshalJSON() ([]byte, error) {
+	type Alias AuthenticatedPaymentState
 	return json.Marshal(&struct {
 		Amount *decimal.Decimal `json:"amount"`
 		*Alias
 	}{
-		Amount: fromIonDecimal(t.Amount),
-		Alias:  (*Alias)(t),
+		Amount: fromIonDecimal(a.Amount),
+		Alias:  (*Alias)(a),
 	})
 }
 
 // UnmarshalJSON - custom unmarshal of transaction type.
-func (t *Transaction) UnmarshalJSON(data []byte) error {
-	type Alias Transaction
+func (a *AuthenticatedPaymentState) UnmarshalJSON(data []byte) error {
```

Comment:
if we get rid of the ion decimal type can we ditch these custom JSON serialization functions?